Formula/libgcrypt.rb

```diff
@@ -15,12 +15,10 @@ class Libgcrypt < Formula
   depends_on "libgpg-error"
 
   def install
-    # Temporary hack to get libgcrypt building on macOS 10.12 and 10.11 with XCode 8.
-    # Seems to be a Clang issue rather than an upstream one, so
-    # keep checking whether or not this is necessary.
-    # Should be reported to GnuPG if still an issue when near stable.
-    # https://github.com/Homebrew/homebrew-core/issues/1957
-    ENV.O1 if DevelopmentTools.clang_build_version == 800
+    # ./jitterentropy-base.c:60:4: error: "The CPU Jitter random number
+    # generator must not be compiled with optimizations. Use the compiler
+    # switch -O0"
+    ENV.O0
```

Comment:
It's a hard failure. Whether or not we can hack it to build the random number generator & optimise everything but this element is a more open question. There's a whole [paper here](http://chronox.de/jent/doc/CPU-Jitter-NPTRNG.html) discussing the optimisation issue outside of this specific project.